src/processgraph/context.js

```diff
@@ -87,14 +112,21 @@ module.exports = class ProcessingContext {
 				});
 			} */
 		}
-		else {
+		else if(format.toLowerCase() === 'json') {
 			var fileName = Utils.generateHash() + "/result-" + Date.now() +  "." + this.translateOutputFormat(format);
 			var p = path.normalize(path.join(this.serverContext.getTempFolder(), fileName));
 			var parent = path.dirname(p);
 			await fse.ensureDir(parent);
 			await fse.writeJson(p, dataCube.getData());
 			return Utils.getApiUrl("/temp/" + fileName);
 		}
+		else{
```

Comment:
That should never be the case as it should be catched in validation. So this can be removed then...

In the best case we never throw errors in execution as this costs money and/or resources. Most errors should be thrown in validation.

Reply:
Okay, I will keep this in my mind!